api/v1beta1/azurecluster_default.go

```diff
@@ -128,6 +130,12 @@ func (c *AzureCluster) setSubnetDefaults() {
 			if subnet.NatGateway.NatGatewayIP.Name == "" {
 				subnet.NatGateway.NatGatewayIP.Name = generateNatGatewayIPName(c.ObjectMeta.Name, subnet.Name)
 			}
+		} else {
+			if !subnet.IsIPv6Enabled() {
```

Comment:
Should this if statement be outside of the `else` block? Looking through the issue details, it looks like the intended behavior is to set NAT Gateway by default if not using IPv6, regardless of whether a name has been provided or not.

Reply:
Thanks @willie-yao for your review! I think put the statement be outside of the `else` block is ok. 
For a subnet, if the NAT gateway name is provided, this means NAT gateway has beed enabled, we don't need to set it again.  If the NAT gateway name doesn't be provided, then I will go to check if it's using IPv6, if not, then I will set NAT gateway by default, otherwise, I will keep the original subnet settings and do nothing.

Reply:
Ah that makes sense, thanks for the detailed explanation!

Reply:
instead of adding an else condition here with another nested if loop, have you considered directly modifying the conditions for which IsNatGatewayEnabled returns true? and setting the name in NAT gateway name in the block above when it's not already set. This would avoid duplicate code of setting the IP name and nested if/else statements.

Reply:
@CecileRobertMichon Thanks for your review! Indeed, there has a duplicate code for setting NAT gateway IP name. But if I modify the condition of IsNatGatewayEnabled and set the name of NAT gateway name if it's not set yet in this func, then I need to call generateNatGatewayName func in type.go file. This will cause an inconsistency from current CAPZ codes in main branch. Although this does not cause any logical problems, it just looks less elegant formally. So I prefer to judge if this is a IPv6 cluster first, then go to judge if IsNatGatewayEnabled or not. This will also avoid duplication. Does this make sense?